src/coreclr/jit/morph.cpp

```diff
@@ -9236,10 +9236,8 @@ GenTree* Compiler::fgMorphCall(GenTreeCall* call)
             fgWalkTreePost(&value, resetMorphedFlag);
 #endif // DEBUG
 
-            GenTree* const nullCheckedArr = impCheckForNullPointer(arr);
```

Comment:
This is the morph bug: it was creating an `ASG` in a statement that did not get properly prepended.